### DISCUSSION TITLE
overlapping named-entity spans

### DISCUSSION THREAD

**Discussion by miwieg** [explosion/spaCy#4966] [New Features & Project Ideas]
For a project I am working on I need to extend the existing named-entity recognition.
I first tried to add new spans and add them to doc.ents. However, this only seems to work if the new entities do not overlap with spans of the original named-entity recognition.
For the application, I have in mind, there will definitely be overlapping spans. (So, it will not work if I remove the original named-entity recognition.)
Is there a work-around for my problem.

## Which page or section is this issue related to?

https://spacy.io/usage/processing-pipelines#custom-components-simple

**Comment by svlandeg** [comment]
The current implementation of the NER module indeed prohibits overlapping entity spans, though this will likely be revisited in the near future.

One option you have is to store the potentially overlapping spans in [extension attributes](https://spacy.io/usage/processing-pipelines#custom-components-attributes). This should give your more flexibility.

**Comment by miwieg** [comment]
Thank you very much for your helpful reply.
What I now did was creating for the additional entity recognizer an extension attribute to doc.
In this extension attribute I store a list of additional spans.

**Comment by svlandeg** [comment]
Great, happy to hear that works for you!
